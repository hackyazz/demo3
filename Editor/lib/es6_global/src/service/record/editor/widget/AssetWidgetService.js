


function getWidget(param) {
  return /* Asset */1;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* Asset */1;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
