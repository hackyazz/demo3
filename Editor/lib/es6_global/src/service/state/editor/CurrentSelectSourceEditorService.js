


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
