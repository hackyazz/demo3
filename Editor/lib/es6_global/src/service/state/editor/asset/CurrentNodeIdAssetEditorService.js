

import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as CurrentNodeIdAssetService$WonderEditor from "../../../record/editor/asset/CurrentNodeIdAssetService.js";
import * as RootTreeAssetEditorService$WonderEditor from "./RootTreeAssetEditorService.js";

function getCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
}

function unsafeGetCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.unsafeGetCurrentNodeId(editorState[/* assetRecord */2]);
}

function clearCurrentNodeId(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.clearCurrentNodeId(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setCurrentNodeId(currentNodeId, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.setCurrentNodeId(currentNodeId, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function couldRemoveCurrentNode(editorState) {
  var match = CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
  if (match !== undefined) {
    return !NodeAssetService$WonderEditor.isIdEqual(match, NodeAssetService$WonderEditor.getNodeId(RootTreeAssetEditorService$WonderEditor.getRootNode(editorState)));
  } else {
    return false;
  }
}

export {
  getCurrentNodeId ,
  unsafeGetCurrentNodeId ,
  clearCurrentNodeId ,
  setCurrentNodeId ,
  couldRemoveCurrentNode ,
  
}
/* CurrentNodeIdAssetService-WonderEditor Not a pure module */
