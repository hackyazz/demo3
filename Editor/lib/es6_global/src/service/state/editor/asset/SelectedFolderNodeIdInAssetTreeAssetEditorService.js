

import * as SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor from "../../../record/editor/asset/SelectedFolderNodeIdInAssetTreeAssetService.js";

function getSelectedFolderNodeIdInAssetTree(editorState) {
  return SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.getSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]);
}

function unsafeGetSelectedFolderNodeIdInAssetTree(editorState) {
  return SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.unsafeGetSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]);
}

function clearSelectedFolderNodeIdInAssetTree(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setSelectedFolderNodeIdInAssetTree(selectedFolderNodeIdInAssetTree, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.setSelectedFolderNodeIdInAssetTree(selectedFolderNodeIdInAssetTree, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getSelectedFolderNodeIdInAssetTree ,
  unsafeGetSelectedFolderNodeIdInAssetTree ,
  clearSelectedFolderNodeIdInAssetTree ,
  setSelectedFolderNodeIdInAssetTree ,
  
}
/* SelectedFolderNodeIdInAssetTreeAssetService-WonderEditor Not a pure module */
