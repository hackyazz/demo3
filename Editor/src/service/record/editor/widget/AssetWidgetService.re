open WidgetType;

let getWidget = () => Asset;

let isWidget = startWidget =>
  switch (startWidget) {
  | None => false
  | Some(startWidget) => startWidget === getWidget()
  };