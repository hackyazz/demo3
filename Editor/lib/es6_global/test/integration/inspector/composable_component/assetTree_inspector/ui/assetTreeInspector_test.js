

import * as Curry from "../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../../../../tool/TestTool.js";
import * as ReactTool$WonderEditor from "../../../../../tool/ui/ReactTool.js";
import * as InspectorTool$WonderEditor from "../../../../../tool/ui/InspectorTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../tool/ReactTestTool.js";
import * as EventListenerTool$WonderEditor from "../../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as AssetWidgetService$WonderEditor from "../../../../../../src/service/record/editor/widget/AssetWidgetService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../../../src/service/state/editor/StateEditorService.js";
import * as MainEditorAssetTool$WonderEditor from "../../../../asset/tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../tool/MainEditorSceneTool.js";
import * as AssetTreeInspectorTool$WonderEditor from "../tool/AssetTreeInspectorTool.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../../../asset/tool/MainEditorAssetTreeTool.js";
import * as MainEditorAssetFolderNodeTool$WonderEditor from "../../../../asset/tool/MainEditorAssetFolderNodeTool.js";
import * as MainEditorAssetTextureNodeTool$WonderEditor from "../../../../asset/tool/MainEditorAssetTextureNodeTool.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "../../../../../../src/service/state/editor/asset/CurrentNodeIdAssetEditorService.js";
import * as MainEditorAssetChildrenNodeTool$WonderEditor from "../../../../asset/tool/MainEditorAssetChildrenNodeTool.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";
import * as SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor from "../../../../../../src/service/state/editor/asset/SelectedFolderNodeIdInAssetTreeAssetEditorService.js";

describe("assetTree inspector", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("prepare currentSelectSource", (function () {
                beforeEach((function () {
                        MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorAssetTool$WonderEditor.initAssetTree);
                        var partial_arg = AssetWidgetService$WonderEditor.getWidget(/* () */0);
                        return StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                                      return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                                    }));
                      }));
                afterEach((function () {
                        StateEditorService$WonderEditor.setState(SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(CurrentNodeIdAssetEditorService$WonderEditor.clearCurrentNodeId(StateEditorService$WonderEditor.getState(/* () */0))));
                        return /* () */0;
                      }));
                describe("test component snapshot", (function () {
                        Wonder_jest.test("if hasn't current node, show nothing", (function () {
                                MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                              }));
                        describe("else", (function () {
                                Wonder_jest.test("test set folder to be current node", (function () {
                                        var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                        MainEditorAssetChildrenNodeTool$WonderEditor.selectFolderNode(Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData), undefined, undefined, /* () */0);
                                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                                      }));
                                return Wonder_jest.test("test set texture to be current node", (function () {
                                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildOneTextureAssetTree */0], /* () */0);
                                              MainEditorAssetChildrenNodeTool$WonderEditor.selectTextureNode(Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData), undefined, undefined, /* () */0);
                                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test rename folder node", (function () {
                        Wonder_jest.test("test rename to specific name", (function () {
                                var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData);
                                AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetFolderNode */3](undefined, undefined, nodeId, "mickeyFolder", /* () */0);
                                MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](nodeId, undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                              }));
                        describe("test the root folder can't be rename", (function () {
                                return Wonder_jest.test("the root treeNode->rename-input->disabled should be true", (function () {
                                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                              var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getRootNodeId */2], assetTreeData);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](AssetTreeInspectorTool$WonderEditor.Rename[/* isFolderNameDisabled */4](nodeId)), true);
                                            }));
                              }));
                        describe("test rename asset tree children node", (function () {
                                describe("if node has ext name", (function () {
                                        Wonder_jest.test("rename input should show it", (function () {
                                                var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                                var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData);
                                                StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                                                        return MainEditorAssetFolderNodeTool$WonderEditor.setFolderName(nodeId, "folder1.aaa", param);
                                                      }));
                                                MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](nodeId, undefined, /* () */0);
                                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                                              }));
                                        return Wonder_jest.test("if rename success, the newName should include ext name", (function () {
                                                      var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                                      var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData);
                                                      AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetFolderNode */3](undefined, undefined, nodeId, "folder.aaa", /* () */0);
                                                      MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](nodeId, undefined, /* () */0);
                                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        describe("deal with specific case", (function () {
                                return Wonder_jest.test("key in '', trigger onBlur, the input value should be original name", (function () {
                                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                              var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData);
                                              var reasonStateUpdate = ReactTool$WonderEditor.getUpdateState(AssetTreeInspectorTool$WonderEditor.reducer(undefined, undefined, nodeId, /* Blur */0, /* record */[
                                                        /* inputValue */"",
                                                        /* originalName */"bbb"
                                                      ], /* () */0));
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](reasonStateUpdate[/* inputValue */0]), "bbb");
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test rename texture node", (function () {
                        describe("test rename asset tree children node", (function () {
                                describe("if node has ext name", (function () {
                                        return Wonder_jest.test("rename input shouldn't show it", (function () {
                                                      var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildOneTextureAssetTree */0], /* () */0);
                                                      var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData);
                                                      StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                                                              return MainEditorAssetTextureNodeTool$WonderEditor.setTextureImageName(nodeId, "texture1.png", param);
                                                            }));
                                                      MainEditorAssetChildrenNodeTool$WonderEditor.selectTextureNode(nodeId, undefined, undefined, /* () */0);
                                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
