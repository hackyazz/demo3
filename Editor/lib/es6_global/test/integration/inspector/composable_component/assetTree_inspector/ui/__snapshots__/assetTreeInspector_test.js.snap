// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assetTree inspector prepare currentSelectSource test component snapshot else test set folder to be current node 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <div
      className="inspector-asset-folder"
    >
      <h1>
        Folder
      </h1>
      <hr />
      <div
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span>
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="NoName Folder"
          />
        </div>
      </div>
    </div>
  </article>
</article>
`;

exports[`assetTree inspector prepare currentSelectSource test component snapshot else test set texture to be current node 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <h1>
        Texture
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="texture1"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            WrapS
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            WrapT
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            Mag Filter
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            Min Filter
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
            <option
              value="2"
            >
              Nearest_mipmap_nearest
            </option>
            <option
              value="3"
            >
              Linear_mipmap_nearest
            </option>
            <option
              value="4"
            >
              Nearest_mipmap_linear
            </option>
            <option
              value="5"
            >
              Linear_mipmap_linear
            </option>
          </select>
        </div>
      </article>
    </article>
  </article>
</article>
`;

exports[`assetTree inspector prepare currentSelectSource test component snapshot if hasn't current node, show nothing 1`] = `
<article
  className="wonder-inspector-component"
/>
`;

exports[`assetTree inspector prepare currentSelectSource test rename folder node test rename asset tree children node if node has ext name if rename success, the newName should include ext name 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/remove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            <div
              className="draggable-container select-active"
              draggable={true}
              onClick={[Function]}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              style={Object {}}
            >
              folder.aaa
            </div>
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  />
</article>
`;

exports[`assetTree inspector prepare currentSelectSource test rename folder node test rename asset tree children node if node has ext name rename input should show it 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <div
      className="inspector-asset-folder"
    >
      <h1>
        Folder
      </h1>
      <hr />
      <div
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span>
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="folder1.aaa"
          />
        </div>
      </div>
    </div>
  </article>
</article>
`;

exports[`assetTree inspector prepare currentSelectSource test rename folder node test rename to specific name 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <div
      className="inspector-asset-folder"
    >
      <h1>
        Folder
      </h1>
      <hr />
      <div
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span>
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="mickeyFolder"
          />
        </div>
      </div>
    </div>
  </article>
</article>
`;

exports[`assetTree inspector prepare currentSelectSource test rename texture node test rename asset tree children node if node has ext name rename input shouldn't show it 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <h1>
        Texture
      </h1>
      <hr />
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="texture1"
          />
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            WrapS
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            WrapT
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="2"
            >
              Repeat
            </option>
            <option
              value="1"
            >
              Mirrored_repeat
            </option>
            <option
              value="0"
            >
              Clamp_to_edge
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            Mag Filter
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="1"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
          </select>
        </div>
      </article>
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className=""
          >
            Min Filter
          </span>
        </div>
        <div
          className="item-content"
        >
          <select
            onChange={[Function]}
            value="0"
          >
            <option
              value="0"
            >
              Nearest
            </option>
            <option
              value="1"
            >
              Linear
            </option>
            <option
              value="2"
            >
              Nearest_mipmap_nearest
            </option>
            <option
              value="3"
            >
              Linear_mipmap_nearest
            </option>
            <option
              value="4"
            >
              Nearest_mipmap_linear
            </option>
            <option
              value="5"
            >
              Linear_mipmap_linear
            </option>
          </select>
        </div>
      </article>
    </article>
  </article>
</article>
`;
